Tsan assert in RabbitMQ/AMQP-cpp encryption

test_storage_rabbitmq/test.py::test_row_based_formats failed in https://s3.amazonaws.com/clickhouse-test-reports/57000/d348db84380c00625a71d2b1d1a5c5276e205fb3/integration_tests__tsan__[4_6].html

There is a high chance this is related to emergency fix
https://github.com/ClickHouse/ClickHouse/pull/56703

PR #57000 included #56703 and #56767.

@devcrafter Want to add another exclude for tsan to https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_storage_rabbitmq/test.py#L113?

```
        if sanitizer_assert_instance is not None:
>           raise Exception(
                "Sanitizer assert found for instance {}".format(
                    sanitizer_assert_instance
                )
            )
E           Exception: Sanitizer assert found for instance ==================
E           WARNING: ThreadSanitizer: data race (pid=449)
E             Read of size 4 at 0x7b24001a1710 by thread T264:
E               #0 AMQP::SslConnected::send(char const*, unsigned long) build_docker/./contrib/AMQP-CPP/src/linux_tcp/sslconnected.h:391:13 (clickhouse+0x1e593523) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #1 AMQP::TcpConnection::onData(AMQP::Connection*, char const*, unsigned long) build_docker/./contrib/AMQP-CPP/src/linux_tcp/tcpconnection.cpp:196:13 (clickhouse+0x1e58bc5f) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #2 AMQP::PassthroughBuffer::flush() build_docker/./contrib/AMQP-CPP/src/passthroughbuffer.h:64:19 (clickhouse+0x1e57f413) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #3 AMQP::PassthroughBuffer::~PassthroughBuffer() build_docker/./contrib/AMQP-CPP/src/passthroughbuffer.h:125:24 (clickhouse+0x1e57f413)
E               #4 AMQP::ConnectionImpl::send(AMQP::Frame const&) build_docker/./contrib/AMQP-CPP/src/connectionimpl.cpp:399:5 (clickhouse+0x1e57f413)
E               #5 AMQP::ChannelImpl::send(AMQP::Frame const&) build_docker/./contrib/AMQP-CPP/src/channelimpl.cpp:804:23 (clickhouse+0x1e57207c) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #6 AMQP::ChannelImpl::ack(unsigned long, int) build_docker/./contrib/AMQP-CPP/src/channelimpl.cpp:698:12 (clickhouse+0x1e57207c)
E               #7 AMQP::Channel::ack(unsigned long, int) build_docker/./contrib/AMQP-CPP/include/amqpcpp/channel.h:552:75 (clickhouse+0x1947a460) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #8 DB::RabbitMQConsumer::ackMessages(DB::RabbitMQConsumer::CommitInfo const&) build_docker/./src/Storages/RabbitMQ/RabbitMQConsumer.cpp:111:30 (clickhouse+0x1947a460)
E               #9 DB::RabbitMQSource::sendAck() build_docker/./src/Storages/RabbitMQ/RabbitMQSource.cpp:260:20 (clickhouse+0x194804dd) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #10 DB::RabbitMQSource::generate() build_docker/./src/Storages/RabbitMQ/RabbitMQSource.cpp:112:9 (clickhouse+0x194804dd)
E               #11 DB::ISource::tryGenerate() build_docker/./src/Processors/ISource.cpp:139:18 (clickhouse+0x19de9fd4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #12 DB::ISource::work() build_docker/./src/Processors/ISource.cpp:108:26 (clickhouse+0x19de9a0a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #13 DB::executeJob(DB::ExecutingGraph::Node*, DB::ReadProgressCallback*) build_docker/./src/Processors/Executors/ExecutionThreadContext.cpp:47:26 (clickhouse+0x19e0c8b4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #14 DB::ExecutionThreadContext::executeTask() build_docker/./src/Processors/Executors/ExecutionThreadContext.cpp:95:9 (clickhouse+0x19e0c8b4)
E               #15 DB::PipelineExecutor::executeStepImpl(unsigned long, std::__1::atomic<bool>*) build_docker/./src/Processors/Executors/PipelineExecutor.cpp:273:26 (clickhouse+0x19dff8b0) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #16 DB::PipelineExecutor::executeSingleThread(unsigned long) build_docker/./src/Processors/Executors/PipelineExecutor.cpp:239:5 (clickhouse+0x19dfead8) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #17 DB::PipelineExecutor::executeImpl(unsigned long, bool) build_docker/./src/Processors/Executors/PipelineExecutor.cpp:399:9 (clickhouse+0x19dfead8)
E               #18 DB::PipelineExecutor::execute(unsigned long, bool) build_docker/./src/Processors/Executors/PipelineExecutor.cpp:111:9 (clickhouse+0x19dfe7de) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #19 DB::threadFunction(DB::PullingAsyncPipelineExecutor::Data&, std::__1::shared_ptr<DB::ThreadGroup>, unsigned long, bool) build_docker/./src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:84:24 (clickhouse+0x19e11438) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #20 DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0::operator()() const build_docker/./src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:112:13 (clickhouse+0x19e11438)
E               #21 decltype(std::declval<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&>()()) std::__1::__invoke[abi:v15000]<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x19e11438)
E               #22 decltype(auto) std::__1::__apply_tuple_impl[abi:v15000]<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&, std::__1::tuple<>&>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&, std::__1::tuple<>&, std::__1::__tuple_indices<>) build_docker/./contrib/llvm-project/libcxx/include/tuple:1789:1 (clickhouse+0x19e11438)
E               #23 decltype(auto) std::__1::apply[abi:v15000]<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&, std::__1::tuple<>&>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&, std::__1::tuple<>&) build_docker/./contrib/llvm-project/libcxx/include/tuple:1798:1 (clickhouse+0x19e11438)
E               #24 ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'()::operator()() build_docker/./src/Common/ThreadPool.h:250:13 (clickhouse+0x19e11438)
E               #25 decltype(std::declval<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>()()) std::__1::__invoke[abi:v15000]<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'()&>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x19e11438)
E               #26 void std::__1::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'()&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:479:9 (clickhouse+0x19e11438)
E               #27 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'(), void ()>::operator()[abi:v15000]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:235:12 (clickhouse+0x19e11438)
E               #28 void std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:16 (clickhouse+0x19e11438)
E               #29 std::__1::__function::__policy_func<void ()>::operator()[abi:v15000]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:848:16 (clickhouse+0xf34576e) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #30 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:1187:12 (clickhouse+0xf34576e)
E               #31 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) build_docker/./src/Common/ThreadPool.cpp:421:13 (clickhouse+0xf34576e)
E               #32 void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()::operator()() const build_docker/./src/Common/ThreadPool.cpp:183:73 (clickhouse+0xf34c311) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #33 decltype(std::declval<void>()()) std::__1::__invoke[abi:v15000]<void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>(void&&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0xf34c311)
E               #34 void std::__1::__thread_execute[abi:v15000]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>(std::__1::tuple<void, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>&, std::__1::__tuple_indices<>) build_docker/./contrib/llvm-project/libcxx/include/thread:284:5 (clickhouse+0xf34c311)
E               #35 void* std::__1::__thread_proxy[abi:v15000]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>>(void*) build_docker/./contrib/llvm-project/libcxx/include/thread:295:5 (clickhouse+0xf34c311)
E           
E             Previous write of size 4 at 0x7b24001a1710 by thread T161 (mutexes: write M0, write M1):
E               #0 AMQP::SslConnected::receive(AMQP::Monitor const&) build_docker/./contrib/AMQP-CPP/src/linux_tcp/sslconnected.h:294:20 (clickhouse+0x1e593cc7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #1 AMQP::SslConnected::process(AMQP::Monitor const&, int, int) build_docker/./contrib/AMQP-CPP/src/linux_tcp/sslconnected.h:369:38 (clickhouse+0x1e593492) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #2 AMQP::TcpConnection::process(int, int) build_docker/./contrib/AMQP-CPP/src/linux_tcp/tcpconnection.cpp:102:30 (clickhouse+0x1e58b5b4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #3 AMQP::LibUvHandler::Watcher::callback(uv_poll_s*, int, int) build_docker/./contrib/AMQP-CPP/include/amqpcpp/libuv.h:69:25 (clickhouse+0x193fe28d) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #4 uv__poll_io build_docker/./contrib/libuv/src/unix/poll.c (clickhouse+0x1e5b949a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #5 uv__io_poll build_docker/./contrib/libuv/src/unix/epoll.c:385:11 (clickhouse+0x1e5bfd74) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #6 uv_run build_docker/./contrib/libuv/src/unix/core.c:406:5 (clickhouse+0x1e5b467e) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #7 DB::RabbitMQHandler::startLoop() build_docker/./src/Storages/RabbitMQ/RabbitMQHandler.cpp:43:9 (clickhouse+0x19408ce8) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #8 DB::StorageRabbitMQ::loopingFunc() build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:270:30 (clickhouse+0x193f6599) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #9 DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0::operator()() const build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:185:107 (clickhouse+0x193f6599)
E               #10 decltype(std::declval<DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0&>()()) std::__1::__invoke[abi:v15000]<DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0&>(DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x193f6599)
E               #11 void std::__1::__invoke_void_return_wrapper<void, true>::__call<DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0&>(DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:479:9 (clickhouse+0x193f6599)
E               #12 std::__1::__function::__default_alloc_func<DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0, void ()>::operator()[abi:v15000]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:235:12 (clickhouse+0x193f6599)
E               #13 void std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool)::$_0, void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:16 (clickhouse+0x193f6599)
E               #14 std::__1::__function::__policy_func<void ()>::operator()[abi:v15000]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:848:16 (clickhouse+0x16d281b7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #15 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:1187:12 (clickhouse+0x16d281b7)
E               #16 DB::BackgroundSchedulePoolTaskInfo::execute() build_docker/./src/Core/BackgroundSchedulePool.cpp:104:9 (clickhouse+0x16d281b7)
E               #17 DB::BackgroundSchedulePool::threadFunction() build_docker/./src/Core/BackgroundSchedulePool.cpp:302:19 (clickhouse+0x16d2b62d) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #18 DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0::operator()() const build_docker/./src/Core/BackgroundSchedulePool.cpp:168:79 (clickhouse+0x16d2bf4a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #19 decltype(std::declval<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&>()()) std::__1::__invoke[abi:v15000]<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x16d2bf4a)
E               #20 decltype(auto) std::__1::__apply_tuple_impl[abi:v15000]<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&, std::__1::tuple<>&>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&, std::__1::tuple<>&, std::__1::__tuple_indices<>) build_docker/./contrib/llvm-project/libcxx/include/tuple:1789:1 (clickhouse+0x16d2bf4a)
E               #21 decltype(auto) std::__1::apply[abi:v15000]<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&, std::__1::tuple<>&>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&, std::__1::tuple<>&) build_docker/./contrib/llvm-project/libcxx/include/tuple:1798:1 (clickhouse+0x16d2bf4a)
E               #22 ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'()::operator()() build_docker/./src/Common/ThreadPool.h:250:13 (clickhouse+0x16d2bf4a)
E               #23 decltype(std::declval<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>()()) std::__1::__invoke[abi:v15000]<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'()&>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x16d2bf4a)
E               #24 void std::__1::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'()&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:479:9 (clickhouse+0x16d2bf4a)
E               #25 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'(), void ()>::operator()[abi:v15000]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:235:12 (clickhouse+0x16d2bf4a)
E               #26 void std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0>(DB::BackgroundSchedulePool::BackgroundSchedulePool(unsigned long, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, StrongTypedef<unsigned long, CurrentMetrics::MetricTag>, char const*)::$_0&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:16 (clickhouse+0x16d2bf4a)
E               #27 std::__1::__function::__policy_func<void ()>::operator()[abi:v15000]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:848:16 (clickhouse+0xf34576e) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #28 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:1187:12 (clickhouse+0xf34576e)
E               #29 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) build_docker/./src/Common/ThreadPool.cpp:421:13 (clickhouse+0xf34576e)
E               #30 void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()::operator()() const build_docker/./src/Common/ThreadPool.cpp:183:73 (clickhouse+0xf34c311) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #31 decltype(std::declval<void>()()) std::__1::__invoke[abi:v15000]<void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>(void&&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0xf34c311)
E               #32 void std::__1::__thread_execute[abi:v15000]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>(std::__1::tuple<void, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>&, std::__1::__tuple_indices<>) build_docker/./contrib/llvm-project/libcxx/include/thread:284:5 (clickhouse+0xf34c311)
E               #33 void* std::__1::__thread_proxy[abi:v15000]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, Priority, std::__1::optional<unsigned long>, bool)::'lambda0'()>>(void*) build_docker/./contrib/llvm-project/libcxx/include/thread:295:5 (clickhouse+0xf34c311)
E           
E             Location is heap block of size 144 at 0x7b24001a1690 allocated by thread T4:
E               #0 operator new(unsigned long) <null> (clickhouse+0x7312bb7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #1 AMQP::SslHandshake::nextstate(AMQP::Monitor const&) build_docker/./contrib/AMQP-CPP/src/linux_tcp/sslhandshake.h:69:29 (clickhouse+0x1e592aa0) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #2 AMQP::SslHandshake::process(AMQP::Monitor const&, int, int) build_docker/./contrib/AMQP-CPP/src/linux_tcp/sslhandshake.h:181:33 (clickhouse+0x1e5927c2) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #3 AMQP::TcpConnection::process(int, int) build_docker/./contrib/AMQP-CPP/src/linux_tcp/tcpconnection.cpp:102:30 (clickhouse+0x1e58b5b4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #4 AMQP::LibUvHandler::Watcher::callback(uv_poll_s*, int, int) build_docker/./contrib/AMQP-CPP/include/amqpcpp/libuv.h:69:25 (clickhouse+0x193fe28d) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #5 uv__poll_io build_docker/./contrib/libuv/src/unix/poll.c (clickhouse+0x1e5b949a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #6 uv__io_poll build_docker/./contrib/libuv/src/unix/epoll.c:385:11 (clickhouse+0x1e5bfd74) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #7 uv_run build_docker/./contrib/libuv/src/unix/core.c:406:5 (clickhouse+0x1e5b467e) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #8 DB::RabbitMQHandler::iterateLoop() build_docker/./src/Storages/RabbitMQ/RabbitMQHandler.cpp:53:16 (clickhouse+0x194091e2) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #9 DB::RabbitMQConnection::connectImpl() build_docker/./src/Storages/RabbitMQ/RabbitMQConnection.cpp:104:23 (clickhouse+0x194052e2) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #10 DB::RabbitMQConnection::connect() build_docker/./src/Storages/RabbitMQ/RabbitMQConnection.cpp:35:5 (clickhouse+0x19404809) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #11 DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool) build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:172:25 (clickhouse+0x193e0ea5) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #12 DB::StorageRabbitMQ* std::__1::construct_at[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, DB::StorageRabbitMQ*>(DB::StorageRabbitMQ*, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/construct_at.h:35:48 (clickhouse+0x19403793) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #13 void std::__1::allocator_traits<std::__1::allocator<DB::StorageRabbitMQ>>::construct[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void, void>(std::__1::allocator<DB::StorageRabbitMQ>&, DB::StorageRabbitMQ*, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/allocator_traits.h:298:9 (clickhouse+0x19403793)
E               #14 std::__1::__shared_ptr_emplace<DB::StorageRabbitMQ, std::__1::allocator<DB::StorageRabbitMQ>>::__shared_ptr_emplace[abi:v15000]<DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&>(std::__1::allocator<DB::StorageRabbitMQ>, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:292:9 (clickhouse+0x19403793)
E               #15 std::__1::shared_ptr<DB::StorageRabbitMQ> std::__1::allocate_shared[abi:v15000]<DB::StorageRabbitMQ, std::__1::allocator<DB::StorageRabbitMQ>, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void>(std::__1::allocator<DB::StorageRabbitMQ> const&, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:953:55 (clickhouse+0x19403793)
E               #16 std::__1::shared_ptr<DB::StorageRabbitMQ> std::__1::make_shared[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void>(DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:962:12 (clickhouse+0x193f96f6) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #17 DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0::operator()(DB::StorageFactory::Arguments const&) const build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:1211:16 (clickhouse+0x193f96f6)
E               #18 decltype(std::declval<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&>()(std::declval<DB::StorageFactory::Arguments const&>())) std::__1::__invoke[abi:v15000]<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&>(DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x193f96f6)
E               #19 std::__1::shared_ptr<DB::IStorage> std::__1::__invoke_void_return_wrapper<std::__1::shared_ptr<DB::IStorage>, false>::__call<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&>(DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:470:16 (clickhouse+0x193f96f6)
E               #20 std::__1::__function::__default_alloc_func<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0, std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()[abi:v15000](DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:235:12 (clickhouse+0x193f96f6)
E               #21 std::__1::shared_ptr<DB::IStorage> std::__1::__function::__policy_invoker<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::__call_impl<std::__1::__function::__default_alloc_func<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0, std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>>(std::__1::__function::__policy_storage const*, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:16 (clickhouse+0x193f96f6)
E               #22 std::__1::__function::__policy_func<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()[abi:v15000](DB::StorageFactory::Arguments const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:848:16 (clickhouse+0x18f75a49) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #23 std::__1::function<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()(DB::StorageFactory::Arguments const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:1187:12 (clickhouse+0x18f75a49)
E               #24 DB::StorageFactory::get(DB::ASTCreateQuery const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, bool) const build_docker/./src/Storages/StorageFactory.cpp:225:16 (clickhouse+0x18f75a49)
E               #25 DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&, std::__1::unique_ptr<DB::DDLGuard, std::__1::default_delete<DB::DDLGuard>>&) build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1425:42 (clickhouse+0x1801f082) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #26 DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1254:20 (clickhouse+0x180170f5) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #27 DB::InterpreterCreateQuery::execute() build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1724:16 (clickhouse+0x18024bb7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #28 DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) build_docker/./src/Interpreters/executeQuery.cpp:1103:40 (clickhouse+0x18836ee4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #29 DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) build_docker/./src/Interpreters/executeQuery.cpp:1281:26 (clickhouse+0x18832897) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #30 DB::TCPHandler::runImpl() build_docker/./src/Server/TCPHandler.cpp:498:54 (clickhouse+0x19d6be52) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #31 DB::TCPHandler::run() build_docker/./src/Server/TCPHandler.cpp:2283:9 (clickhouse+0x19d89ca7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #32 Poco::Net::TCPServerConnection::start() build_docker/./base/poco/Net/src/TCPServerConnection.cpp:43:3 (clickhouse+0x1d8856e2) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #33 Poco::Net::TCPServerDispatcher::run() build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:115:20 (clickhouse+0x1d885f5a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #34 Poco::PooledThread::run() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:188:14 (clickhouse+0x1da851c6) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #35 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1da8348f) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #36 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1da81949) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E           
E             Mutex M0 (0x7b3400209c08) created at:
E               #0 pthread_mutex_lock <null> (clickhouse+0x728e66b) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #1 std::__1::__libcpp_mutex_lock[abi:v15000](pthread_mutex_t*) build_docker/./contrib/llvm-project/libcxx/include/__threading_support:304:10 (clickhouse+0x20280dd9) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #2 std::__1::mutex::lock() build_docker/./contrib/llvm-project/libcxx/src/mutex.cpp:38:14 (clickhouse+0x20280dd9)
E               #3 std::__1::lock_guard<std::__1::mutex>::lock_guard[abi:v15000](std::__1::mutex&) build_docker/./contrib/llvm-project/libcxx/include/__mutex_base:94:27 (clickhouse+0x16d27c48) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #4 DB::BackgroundSchedulePoolTaskInfo::deactivate() build_docker/./src/Core/BackgroundSchedulePool.cpp:49:21 (clickhouse+0x16d27c48)
E               #5 DB::StorageRabbitMQ::StorageRabbitMQ(DB::StorageID const&, std::__1::shared_ptr<DB::Context const>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool) build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:186:19 (clickhouse+0x193e11a1) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #6 DB::StorageRabbitMQ* std::__1::construct_at[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, DB::StorageRabbitMQ*>(DB::StorageRabbitMQ*, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/construct_at.h:35:48 (clickhouse+0x19403793) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #7 void std::__1::allocator_traits<std::__1::allocator<DB::StorageRabbitMQ>>::construct[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void, void>(std::__1::allocator<DB::StorageRabbitMQ>&, DB::StorageRabbitMQ*, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/allocator_traits.h:298:9 (clickhouse+0x19403793)
E               #8 std::__1::__shared_ptr_emplace<DB::StorageRabbitMQ, std::__1::allocator<DB::StorageRabbitMQ>>::__shared_ptr_emplace[abi:v15000]<DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&>(std::__1::allocator<DB::StorageRabbitMQ>, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:292:9 (clickhouse+0x19403793)
E               #9 std::__1::shared_ptr<DB::StorageRabbitMQ> std::__1::allocate_shared[abi:v15000]<DB::StorageRabbitMQ, std::__1::allocator<DB::StorageRabbitMQ>, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void>(std::__1::allocator<DB::StorageRabbitMQ> const&, DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:953:55 (clickhouse+0x19403793)
E               #10 std::__1::shared_ptr<DB::StorageRabbitMQ> std::__1::make_shared[abi:v15000]<DB::StorageRabbitMQ, DB::StorageID const&, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>, bool const&, void>(DB::StorageID const&, std::__1::shared_ptr<DB::Context>&&, DB::ColumnsDescription const&, std::__1::unique_ptr<DB::RabbitMQSettings, std::__1::default_delete<DB::RabbitMQSettings>>&&, bool const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:962:12 (clickhouse+0x193f96f6) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #11 DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0::operator()(DB::StorageFactory::Arguments const&) const build_docker/./src/Storages/RabbitMQ/StorageRabbitMQ.cpp:1211:16 (clickhouse+0x193f96f6)
E               #12 decltype(std::declval<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&>()(std::declval<DB::StorageFactory::Arguments const&>())) std::__1::__invoke[abi:v15000]<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&>(DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:394:23 (clickhouse+0x193f96f6)
E               #13 std::__1::shared_ptr<DB::IStorage> std::__1::__invoke_void_return_wrapper<std::__1::shared_ptr<DB::IStorage>, false>::__call<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&>(DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0&, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/invoke.h:470:16 (clickhouse+0x193f96f6)
E               #14 std::__1::__function::__default_alloc_func<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0, std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()[abi:v15000](DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:235:12 (clickhouse+0x193f96f6)
E               #15 std::__1::shared_ptr<DB::IStorage> std::__1::__function::__policy_invoker<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::__call_impl<std::__1::__function::__default_alloc_func<DB::registerStorageRabbitMQ(DB::StorageFactory&)::$_0, std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>>(std::__1::__function::__policy_storage const*, DB::StorageFactory::Arguments const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:16 (clickhouse+0x193f96f6)
E               #16 std::__1::__function::__policy_func<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()[abi:v15000](DB::StorageFactory::Arguments const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:848:16 (clickhouse+0x18f75a49) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #17 std::__1::function<std::__1::shared_ptr<DB::IStorage> (DB::StorageFactory::Arguments const&)>::operator()(DB::StorageFactory::Arguments const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:1187:12 (clickhouse+0x18f75a49)
E               #18 DB::StorageFactory::get(DB::ASTCreateQuery const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, std::__1::shared_ptr<DB::Context>, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, bool) const build_docker/./src/Storages/StorageFactory.cpp:225:16 (clickhouse+0x18f75a49)
E               #19 DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&, std::__1::unique_ptr<DB::DDLGuard, std::__1::default_delete<DB::DDLGuard>>&) build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1425:42 (clickhouse+0x1801f082) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #20 DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1254:20 (clickhouse+0x180170f5) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #21 DB::InterpreterCreateQuery::execute() build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:1724:16 (clickhouse+0x18024bb7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #22 DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) build_docker/./src/Interpreters/executeQuery.cpp:1103:40 (clickhouse+0x18836ee4) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #23 DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) build_docker/./src/Interpreters/executeQuery.cpp:1281:26 (clickhouse+0x18832897) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #24 DB::TCPHandler::runImpl() build_docker/./src/Server/TCPHandler.cpp:498:54 (clickhouse+0x19d6be52) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #25 DB::TCPHandler::run() build_docker/./src/Server/TCPHandler.cpp:2283:9 (clickhouse+0x19d89ca7) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #26 Poco::Net::TCPServerConnection::start() build_docker/./base/poco/Net/src/TCPServerConnection.cpp:43:3 (clickhouse+0x1d8856e2) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #27 Poco::Net::TCPServerDispatcher::run() build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:115:20 (clickhouse+0x1d885f5a) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #28 Poco::PooledThread::run() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:188:14 (clickhouse+0x1da851c6) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #29 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1da8348f) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E               #30 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1da81949) (BuildId: 7d9d20953a67635ca82fad14878b820761f56633)
E           
```

--- a/tests/integration/test_storage_rabbitmq/test.py
+++ b/tests/integration/test_storage_rabbitmq/test.py
@@ -110,7 +110,7 @@ def rabbitmq_setup_teardown():
     ],
 )
 def test_rabbitmq_select(rabbitmq_cluster, secure):
-    if secure and instance.is_built_with_memory_sanitizer():
+    if secure and instance.is_built_with_thread_sanitizer():
         pytest.skip(
             "Data races: see https://github.com/ClickHouse/ClickHouse/issues/56866"
         )

